src/Microsoft.JSInterop/IJSInProcessRuntime.cs

```diff
@@ -0,0 +1,20 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.JSInterop
+{
+    /// <summary>
+    /// Represents an instance of a JavaScript runtime to which calls may be dispatched.
+    /// </summary>
+    public interface IJSInProcessRuntime : IJSRuntime
+    {
+        /// <summary>
+        /// Invokes the specified JavaScript function synchronously.
+        /// </summary>
+        /// <typeparam name="T">The JSON-serializable return type.</typeparam>
+        /// <param name="identifier">An identifier for the function to invoke. For example, the value <code>"someScope.someFunction"</code> will invoke the function <code>window.someScope.someFunction</code>.</param>
+        /// <param name="args">JSON-serializable arguments.</param>
+        /// <returns>An instance of <typeparamref name="T"/> obtained by JSON-deserializing the return value.</returns>
+        T Invoke<T>(string identifier, params object[] args);
```

Comment:
Would about void-returning methods? Is the expectation that you'd use `object` as the type?

Reply:
or (`System.Void`)

Reply:
Yes, we have been using `<object>` for that (the compiler doesn't allow `void` or `System.Void`). I'm aware it's a bit clunky and we can add another overload for that if people turn out to want it.